src/JobPipeline.php

```diff
@@ -63,7 +64,18 @@ public function handle(): void
                 $job = [new $job(...$this->passable), 'handle'];
             }
 
-            $result = app()->call($job);
+            try {
+                $result = app()->call($job);
+            } catch(Throwable $exception) {
+                if ( method_exists(get_class($job[0]), 'failed') ) {
+                    
+                    call_user_func_array([$job[0], 'failed'], [$exception]);
```

Comment:
Not sure if this is the right approach. We should be using Laravel's queue logic for this, rather than checking method existence I think.

Perhaps you can take a look at the Laravel source code to see how the `failed` methods are invoked.

Reply:
Looks like this is essentially how Laravel handles this same problem:

[`if (method_exists($this->instance = $this->resolve($class), 'failed')) {`](https://github.com/laravel/framework/blob/c0f3ee8c8779d5cfb6bb3c37c93b84672e67f236/src/Illuminate/Queue/Jobs/Job.php#L212)

[`if (method_exists($command, 'failed')) {`](https://github.com/laravel/framework/blob/6c0ffe3b274aeff16661efc33921ae5211b5f7d2/src/Illuminate/Queue/CallQueuedHandler.php#L262)

Not sure if there's a better way to do this?